test/unit/queue/src/QueueTest.cpp

```diff
@@ -101,8 +85,11 @@ TEMPLATE_LIST_TEST_CASE(
         f.m_queue,
         [&f, &callbackFinished]() noexcept
         {
-            LOOPED_CHECK(30, 100, !alpaka::empty(f.m_queue));
+            // This callback function is in the queue therefore the queue can not be empty.
+            CHECK(!alpaka::empty(f.m_queue));
             std::this_thread::sleep_for(std::chrono::milliseconds(100u));
+            // Check again, the queue must stay not empty.
+            CHECK(!alpaka::empty(f.m_queue));
```

Comment:
So this is just a simplification of the test without a change in behavior?

Reply:
No this is extending the test, we need to check that the queue stays non-empty as long as the task within the callback queue is executed.

Reply:
The previous test did:
```
CHECK(!empty(queue));
wait 100ms
CHECK(!empty(queue));
wait 100ms
CHECK(!empty(queue));
wait 100ms
CHECK(!empty(queue));
wait 100ms
... // 30 times
```
The new test does only:
```
CHECK(!empty(queue));
wait 100ms
CHECK(!empty(queue));
```
IMO this is a simplification. What additional behavior is your new version covering?

Reply:
The previous implementation was only repeating the command in the case where the first check returned that the queue is empty but we are in the host function which is blocking the queue therefore it must be `!empty` else something is wrong.
The second check validates CUDA/HIP or our bookkeeping with the callback thread that if a task which is blocking the execution on the host is really blocking the queue too.

Reply:
Oh yes, you are completely right! Sorry, my mistake! So yes, your proposed change is better than what we had before!